H5_sampler/sample_signal.py

```diff
@@ -0,0 +1,64 @@
+from Sampler import BlackBox, MaxMultiSampler
+from optparse import OptionParser
+import os
+from os.path import join, dirname, exists
+import numpy as np
+import json
+
+parser = OptionParser()
+parser.add_option("-s", "--seed", type=int, default=123, help="Random seed")
+parser.add_option("-i", "--input", type=str,
+                  default="/eos/cms/store/group/phys_b2g/CASE/h5_files/UL/",
+                  help="Input base directory.")
+parser.add_option("-o", "--output", type=str, default="./mixed_years/",
+                  help="Output directory.")
+parser.add_option("-d", "--subdir", type=str, default="new/",
+                  help=("Subdirectory after the year-specific folder "
+                        "(<input base dir>/<year dir>/<subdir>/<actual h5 input samples>)."
+                        "Provide empty string if subdir should be omitted."))
+parser.add_option("-c", "--config", type=str,
+                  default=join(dirname(__file__), "sample_signal_config.json"),
+                  help="JSON config file containing zears and relative lumies.")
+options, args = parser.parse_args()
+
+
+base_dir = options.input
+out_dir = options.output
+subdir = options.subdir
+with open(options.config) as json_file:
+    config = json.load(json_file)
+    years = config["years"]
+    lumies = config["lumies"]
+np.random.seed(options.seed)
+
+assert len(years) > 1
+for year in years:
+    assert year in lumies.keys()
+
+if not exists(out_dir):
+    os.makedirs(out_dir)
+
+h5_files = {}
+for year in years:
+    h5_files[year] = [x for x in os.listdir(join(base_dir, year, subdir))
+                      if x.endswith(".h5")]
+
+signal_file_list = []
+for candidate in h5_files[years[0]]:
+    add_candidate = True
+    for year in years[1:]:
+        if candidate not in h5_files[year]:
+            add_candidate = False
+    if add_candidate:
+        signal_file_list.append(candidate)
```

Comment:
Can be heavily simplified with `set`:
```suggestion
# Find signals present in all years
signal_file_list = set.intersection(*h5_files.values())
```
(note that it's no longer a list, but it doesn't need be)

Reply:
Doesn't work out of the box, so I'd rather keep it as is 😄

Reply:
Oh I see, would need `{}` instead of `[]` for the comprehension on line 43

Reply:
Ok I changed it just because I want you to be happy. I did convert it into list and sort it, though. I just do this for deterministic order, which helps if the execution is aborted in the middle and one wants to restart from there.